### ISSUE TITLE
Add support for Task<T>.Result usage

### ISSUE THREAD

**Issue by JimiC** [OneDrive/onedrive-sdk-csharp#68]
Currently the SDK doesn't support the use of the `Result` property of an `async Task<T>`.
When used it results to a deadlock.
Consider the use of `ConfigureAwait(false)` where applicable (mainly everywhere).

P.S.: When I used `ConfigureAwait(false)` in all steps of a `GetAsync()` call it worked. I know this isn't an easy undertaking but it should be done eventually.

Justification: https://youtu.be/H4EmfpsYcfs?t=3966

**Comment by JimiC** [comment]
@ginach 
Any updates on this? I would really like to see this fixed and released so to remove the special code I'm using to make it work.

**Comment by ginach** [comment]
Thanks for reporting this. We need to talk this over internally to figure out the right thing to do here but feel free to submit a PR if you have specific changes in mind.

**Comment by ificator** [comment]
Since this library can be invoked from UI threads, but does not really care about the SynchronizationContext (i.e. any thread pool thread is fine - we don't need the callbacks on the UI thread), we really should be using `ConfigureAwait(false)`.

**Comment by JimiC** [comment]
I meant to provide a PR for this but got involved in other projects, which kept me *really* busy.

**Comment by ginach** [comment]
Unfortunately, this change will require changes to our code generation logic otherwise changes will be overwritten the next time we generate from the service.

We do want to support this. I had to focus on the MS Graph SDK for a bit and hadn't been able to address the issue here until now. We're actually planning a V2 of the SDK to take in some breaking changes and I had been planning on making this fix then. That said, I can probably make this change sooner since it's not breaking.

As an aside, if the OneDrive functionality you're using is currently supported by [MS Graph](https://graph.microsoft.io) you might want to check out that [library](https://github.com/microsoftgraph/msgraph-sdk-dotnet). I made sure to build Task.Result support into it from the beginning.

**Comment by JimiC** [comment]
Fare enough.
From what I saw on `MS Graph` I'm afraid it lacks the functionality we need for the project.
Never the less, this issue isn't a blocker atm as I have dumped the flow to another thread so to not block the UI thread.
On top of that we are atm on "Escrow" mode (as you there in MS call it) as we are about to release the next version.
Still glad to see that this is taken under consideration for V2.
